web/src/main/java/org/perfrepo/web/adapter/TestAdapter.java

```diff
@@ -0,0 +1,161 @@
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    TestDto createTest(TestDto test);
+
+    /**
+     * Update a {@link TestDto} object.
+     *
+     * @param test Parameters of the test that will be updated.
+     * @return Updated {@link TestDto} object.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the object does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.ValidationException If the input parameters are not valid.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    TestDto updateTest(TestDto test);
+
+    /**
+     * Delete a {@link TestDto} object.
+     *
+     * @param id The test identifier.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the object does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    void deleteTest(Long id);
+
+    /**
+     * Return all tests.
+     *
+     * @return List of all tests.
+     */
+    List<TestDto> getAllTests();
+
+    /**
+     * Return all {@link TestDto} tests that satisfy search conditions.
+     *
+     * @return List of {@link TestDto} tests.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    SearchResult searchTests(TestSearchParams searchParams);
+
+    /**
+     * Add a metric to existing test. If a metric does not exist, it will be created.
+     *
+     * @param metric Metric {@link org.perfrepo.dto.metric.MetricDto} that will be added to the test.
+     * @param testId Test {@link TestDto} identifier.
+     * @return Metric {@link MetricDto} that was added to test.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    MetricDto addMetricToTest(MetricDto metric, Long testId);
+
+    /**
+     * Remove a metric from existing test.
+     *
+     * @param metricId Identifier of {@link org.perfrepo.dto.metric.MetricDto} metric that will be removed.
+     * @param testId Test {@link TestDto} identifier.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    void removeMetricFromTest(Long metricId, Long testId);
+
+    /**
+     * Subscribe the user to the test alerts.
+     *
+     * @param testId Test {@link TestDto} identifier.
+     * @param userId User {@link UserDto} identifier.
+     * @throws org.perfrepo.web.adapter.exceptions.NotFoundException If the test or user does not exist.
+     * @throws org.perfrepo.web.adapter.exceptions.AdapterException If anything bad happened.
+     */
+    void subscribeToTestAlerts(Long testId, Long userId);
```

Comment:
OK, I will also rename method addMetricToTest, removeMetricFromTest to addMetric, removeMetric.